### ISSUE TITLE
Is this working?

### ISSUE THREAD

**Issue by milesigel** [Asthestarsfalll/Symbolic-Music-Genre-Transfer-with-CycleGAN-for-pytorch#1]
Hi,

I saw that you were trying to work on converting this into PyTorch based on your comment on the tf version by the authors. How is this going? Is it operational?

**Comment by Asthestarsfalll** [comment]
I have trained the network 30 times and have obtained several `.pth` files, however I have not tested it yet

**Comment by Asthestarsfalll** [comment]
I have trained the network for 30 epochs and have obtained several `.pth` files, however I have not tested them yet

**Comment by milesigel** [comment]
Are you planning on writing testing code for it or are familiar with a previous version that has testing code? Thanks for writing this btw... much more comprehendible.

**Comment by Asthestarsfalll** [comment]
The testing code will be upload soon

**Comment by Asthestarsfalll** [comment]
The testing code will be uploaded soon

**Comment by Asthestarsfalll** [comment]
Testing code is uploaded, and I think the model files you have can't use now because I change the way of saving network's parameters.You can use my model files which can be downloaded in readme.

**Comment by milesigel** [comment]
Hi, thanks for uploading the test code. Whenever I try to use the test file my outputs are not working. For my orgin.mid file, it works correctly but for the other two, they don't work. Any idea why this might be? Was the whole pipeline working for you? I am using my own dataset which might be the source of the issue so I am just trying to troubleshoot without having to run the code again on the sample dataset.  

My reasoning is that if I can get the orign.mid file correct then there was not something wrong with my NumPy arrays that I used as inputs so that's why I am confused.

**Comment by Asthestarsfalll** [comment]
> Hi, thanks for uploading the test code. Whenever I try to use the test file my outputs are not working. For my orgin.mid file, it works correctly but for the other two, they don't work. Any idea why this might be? Was the whole pipeline working for you? I am using my own dataset which might be the source of the issue so I am just trying to troubleshoot without having to run the code again on the sample dataset.

So am i, only the origin.midi works correctly, the other two are empty.I checked for the output and found that  they're too small(only 10e-16 to 10e-24), so they will be change to 0 when saving. Actually, the author's own code also has the same situation that some outputs are empty.I think maybe the network are not trained well for GAN are hard to train.

**Comment by Asthestarsfalll** [comment]
> Hi, thanks for uploading the test code. Whenever I try to use the test file my outputs are not working. For my orgin.mid file, it works correctly but for the other two, they don't work. Any idea why this might be? Was the whole pipeline working for you? I am using my own dataset which might be the source of the issue so I am just trying to troubleshoot without having to run the code again on the sample dataset.

When I have spare time , I will check to see whether there's something wrong with my code.By the way, thank you for staring my repo！

**Comment by milesigel** [comment]
> > Hi, thanks for uploading the test code. Whenever I try to use the test file my outputs are not working. For my orgin.mid file, it works correctly but for the other two, they don't work. Any idea why this might be? Was the whole pipeline working for you? I am using my own dataset which might be the source of the issue so I am just trying to troubleshoot without having to run the code again on the sample dataset.
> 
> When I have spare time , I will check to see whether there's something wrong with my code.By the way, thank you for staring my repo！

Is there a constraint on the GAN that only produces binary values b/c I only see the sigmoid as the output of the Generator? Could this be the problem or are we meant to assume that the Generator should recognize that the only valid outputs are binary?

**Comment by Asthestarsfalll** [comment]
> > > Hi, thanks for uploading the test code. Whenever I try to use the test file my outputs are not working. For my orgin.mid file, it works correctly but for the other two, they don't work. Any idea why this might be? Was the whole pipeline working for you? I am using my own dataset which might be the source of the issue so I am just trying to troubleshoot without having to run the code again on the sample dataset.
> > 
> > 
> > When I have spare time , I will check to see whether there's something wrong with my code.By the way, thank you for staring my repo！
> 
> Is there a constraint on the GAN that only produces binary values b/c I only see the sigmoid as the output of the Generator? Could this be the problem or are we meant to assume that the Generator should recognize that the only valid outputs are binary?

Sorry ,there is a mistake in  `Class CycleGAN`
![image](https://user-images.githubusercontent.com/72954905/127599361-89d14cf2-d0b3-4eac-8fa0-1d6eaac7a6e9.png)
try to use this to generate gaussian noise, the previous way changes the value of inputs 
As for the binary values , the author's code has the corresponding functions
